### PR TITLE
Changed calculation of avellaneda_market_making parameters to happen at order proposal

### PR DESCRIPTION
**Pull request by keithbaum**
- Changed calculation of avellaneda_market_making parameters to happen only when order_proposal is created instead of every tick. This reduces computation cost (for hummingsim backtesting)
- Added __init__ file to trailing_indicators folder

**Before submitting this PR, please make sure**:

- [x] Your code builds clean without any errors or warnings
- [x] You are using approved title ("feat/", "fix/", "docs/", "refactor/")

**A description of the changes proposed in the pull request**:

### CODE REVIEW

**Review by RC-13 — APPROVED** [review]
LGTM, no issue encountered during testing.

File: hummingbot/strategy/avellaneda_market_making/avellaneda_market_making.pyx [hunk 1]
```diff
@@ -568,7 +569,7 @@ cdef class AvellanedaMarketMakingStrategy(StrategyBase):
             # KAPPA
             # Want the maximum possible spread but with restrictions to avoid negative kappa or division by 0
             max_spread_around_reserved_price = max_spread * (2-self._inventory_risk_aversion) + min_spread * self._inventory_risk_aversion
-            if max_spread_around_reserved_price <= self._gamma * (vol ** 2):
+            if (max_spread_around_reserved_price * self._gamma - (vol * self._gamma) **2) <= s_decimal_zero:
```
Comment (CrimsonJacket) [inline hummingbot/strategy/avellaneda_market_making/avellaneda_market_making.pyx:572]: Is there a reason for this change? Mathematically they are the same.

**Review by CrimsonJacket — APPROVED** [review]
Just one quick question regarding the if-condition for the calculation of kappa. 
Also, including the `__init__.py` file in PR #3194 would be more appropriate.
Other than that, LGTM.

File: hummingbot/strategy/avellaneda_market_making/avellaneda_market_making.pyx (hunk 1 above)
Comment (keithbaum) [reply hummingbot/strategy/avellaneda_market_making/avellaneda_market_making.pyx:572]: Decimal minimum possible number was sometimes mapped to 0 when multiplying by gamma**2. That condition was not emulated when using the original condition. By using this condition which is exactly the same calculation made in the denominator, this situations are caught

**Review by CrimsonJacket — APPROVED** [review]
👍

**Pull request merged**